### DISCUSSION TITLE
Is it possible that "atom" property change iteratively leads to objects reaction?

### DISCUSSION THREAD

**Discussion by ColaWIce** [codewriter-packages/UniMob#28] [Q&A] [answered]
First of all, fantastic work! 

For the following code, is there any way to trigger the reaction?  We want the object tracking its property change.

```
public class Test : LifetimeMonoBehaviour
{
    public Button Button1;
    [Atom] public HolderA Outer { get; set; }
    public class HolderA: ILifetimeScope
    {
        [Atom] public HolderB Inner { get; set; }
        public Lifetime Lifetime { get; }
    }

    public class HolderB : ILifetimeScope
    {
        [Atom] public int TestValue { get; set; } = 0;
        public Lifetime Lifetime { get; }
    }
    protected override void Start()
    {
        Button1.onClick.AddListener(() => Outer.Inner.TestValue++);
        Atom.Reaction(Lifetime, () => Outer,
             value => { Debug.Log($"changed {value}"); }, null, false);
    }
}
```

**Comment by vanifatovvlad** [comment, marked as answer]
No, It's Immpossible. The library operates with separate properties and knows nothing about the objects in which these properties are stored

**Reply by ColaWIce** [reply]
I used reflection to call every Atom properties' get() function of a complex class, layer by layer. Not that good, but it works for me.

**Comment by ColaWIce** [comment]
By the way, could you explain more about atom states, options? Also, I am still confused about parents, children and subscribers.
